service/src/main/java/cz/muni/fi/service/ComponentServiceImpl.java

```diff
@@ -18,89 +17,53 @@
 
     @Override
     public List<Component> listAllComponents() {
-        try {
-            return componentDao.listAllComponents();
-        } catch (Throwable e) {
-            throw new TeamFormulaDataAccessException("Could not list all components.", e);
-        }
+        return componentDao.listAllComponents();
```

Comment:
V zadani bolo napisane ze to ma odchytit vynimku a vratit data access exception

Reply:
Make sure that DataAccessException or its subclass is thrown in case of any exception on a the DAO layer; toto bolo v zadani. oni to maju v tom priklade tak isto ako ja. a org.springframework.dao.InvalidDataAccessApiUsageException je DataAccessException

Reply:
mne je jasne ze to extenduje tu exception ale neviem zo zadania mi vyplyva ze ak dao vypluje vynimku zabal ju do DataAccess a takto to pochopi 90% ludi a bude za to ztrhavat body podla mna

Reply:
No ale ked to je v priklade takto a ono to spravi same od seba, tak naco by som pisal nieco, co uz already niekto spravil

Reply:
ale dobre, mozme to spravit tak ako to bolo..

Reply:
mne to je jasne, ja som len vychadzal zo zadania. urob tak ako ti pride logickejsie

Reply:
Asi to nechajme takto. dal som reset hard a vratil som to. pozrime este ostatne timy, ak to maju vsetci tak ako my, tak to nechajme tak.

Reply:
ked tak to vecer refactornem.